src/sisl/unit/base.py

```diff
@@ -424,3 +426,20 @@ def __call__(self, *units):
 
 # Create base sisl unit conversion object
 units = UnitParser(unit_table)
+
+
+@set_module("sisl.unit")
+def units2dict(func):
+    "Decorator to force units parameter into dictionary"
+
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        # Check for units in kwargs, and adjust as needed
+        if "units" in kwargs and not isinstance(kwargs["units"], dict):
+            unit = kwargs["units"]
```

Comment:
it could still be a list, say 'units=("eV", "Ang")` there would be no ambiguity for the end-user.

Reply:
OK, and what should happen if the user provides `units=("eV", "J")`?

Reply:
an error should be raised, then they should be explicit if they should belong to different dimensions.